src/main/java/sirius/db/jdbc/StatementCompiler.java

```diff
@@ -106,109 +110,107 @@ private void parseSection(String originalSQL, String sql, List<Object> params, C
+
+        if (nextSubstitution.getSecond() || paramValue == null) {
+            tempParams.add(paramValue);
+        } else {
+            tempParams.add(addSQLWildcard(paramValue.toString().toLowerCase(), true));
+        }
+
+        sqlBuilder.append(sql.substring(0, nextSubstitution.getFirst()));
+
+        appendPlaceholdersToStatement(sqlBuilder, paramValue);
+
+        return compileSectionPart(sql.substring(endIndex + 1),
+                                  tempParams,
+                                  sqlBuilder,
+                                  appendToStatement || isParameterFilled(paramValue));
+    }
+
+    private Object computeEffectiveParameterValue(String fullParameterName) throws SQLException {
+        String accessPath = null;
+        String parameterName = fullParameterName;
+        if (fullParameterName.contains(".")) {
+            accessPath = parameterName.substring(parameterName.indexOf(".") + 1);
+            parameterName = parameterName.substring(0, parameterName.indexOf("."));
+        }
+
+        Object paramValue = context.get(parameterName);
+        if (accessPath == null || paramValue == null) {
+            return paramValue;
+        }
+
+        try {
+            return Reflection.evalAccessPath(accessPath, paramValue);
+        } catch (Exception e) {
+            throw new SQLException(NLS.fmtr("StatementCompiler.cannotEvalAccessPath")
+                                      .set("name", parameterName)
+                                      .set("path", accessPath)
+                                      .set("value", paramValue)
+                                      .set("query", originalSQL)
+                                      .format(), e);
+        }
+    }
+
+    private void appendPlaceholdersToStatement(StringBuilder sqlBuilder, Object paramValue) {
+        if (paramValue instanceof Collection<?>) {
+            for (int i = 0; i < ((Collection<?>) paramValue).size(); i++) {
+                if (i > 0) {
+                    sqlBuilder.append(",");
                 }
-            } else {
                 sqlBuilder.append(" ? ");
             }
-            sql = sql.substring(endIndex + 1);
-            index = getNextRelevantIndex(sql);
-            directSubstitution = (index > 0) && (sql.charAt(index) == '$');
-        }
-        sqlBuilder.append(sql);
-        if (parameterFound || !ignoreIfParametersNull) {
-            sb.append(sqlBuilder.toString());
-            params.addAll(tempParams);
+        } else {
+            sqlBuilder.append(" ? ");
         }
     }
 
-    private int findClosingCurlyBracket(String originalSQL, String sql, int index) throws SQLException {
+    private boolean isParameterFilled(Object paramValue) {
+        return (Strings.isFilled(paramValue)) && (!(paramValue instanceof Collection<?>)
+                                                  || !((Collection<?>) paramValue).isEmpty());
```

Comment:
maybe refactor to `Strings.isFilled(paramValue)) && !(paramValue instanceof Collection && ((Collection<?>) paramValue).isEmpty())`